web/src/components/game/goals/GoalManagement.tsx

```diff
@@ -35,8 +37,38 @@ export default function GoalManagement() {
     const { showDetails } = settings;
 
     const [goalUploadOpen, setGoalUploadOpen] = useState(false);
+    const [loading, setLoading] = useState(false); // State to manage loading during deletion
+    const [deleteConfirmationText, setDeleteConfirmationText] = useState("");
+    const settingsDialogRef = useRef<DialogRef>(null); // Settings dialog
+    const deleteDialogRef = useRef<DialogRef>(null); // Delete confirmation dialog
 
-    const dialogRef = useRef<DialogRef>(null);
+    // Function to handle deleting all goals
+    const deleteAllGoals = async () => {
+        if (deleteConfirmationText !== "DELETE") {
+            // Handle failure silently
+            return;
+        }
+    
+        try {
+            setLoading(true);
+            const response = await fetch(`/api/games/${slug}/delete-all-goals`, {
+                method: 'DELETE',
+            });
+    
+            if (!response.ok) {
+                // Handle failure silently
```

Comment:
This failure doesn't necessarily need to be silent, you can toast the message sent by the server if you feel it is appropriate (and safe) to do so. Generally, failure cases where the failure is not obvious should not fail silently